Outposts: clarify unmanaged tag requirement

jeana-redhat opened the pull request:
Version(s):
4.15+

Issue:
N/A, via [Slack chat](https://redhat-internal.slack.com/archives/C06K5PQSJ2H/p1709128411762009?thread_ts=1708983737.498759&cid=C06K5PQSJ2H)

Link to docs preview:
* [AWS Outposts on OpenShift Container Platform requirements and limitations](https://72291--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-aws-outposts.html#aws-outposts-requirements-limitations_configuring-aws-outposts)
* [CloudFormation template for the VPC subnet](https://72291--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/configuring-aws-outposts.html#installation-cloudformation-subnet-localzone_configuring-aws-outposts)

QE review:
- [x] QE has approved this change.

Additional information:
Post-GA refinement

mtulio reviewed (commented):
/lgtm

mtulio commented:
I'd also suggest adding some subsection in the begging of the book stating some key points used in the deployment to remediation some AWS Outpost limitations. Those are:
- "unmanaged" tag: it is preventing CCM to choose subnets on OP when creating services in certain situations. A Bug has been filled to work w/ upstream on this: https://issues.redhat.com/browse/OCPBUGS-30041
- new machine/node labels and taint: the taint applied to the node is preventing spreading regular cluster workloads to the Outpost nodes, saving extra configuration like changing the default CSI, and also dedicating the AWS Oupost infrastructure to the user-workloads

jeana-redhat commented:
> I'd also suggest adding some subsection in the begging of the book stating some key points used in the deployment to remediation some AWS Outpost limitations. Those are:
> 
>     * "unmanaged" tag: it is preventing CCM to choose subnets on OP when creating services in certain situations. A Bug has been filled to work w/ upstream on this: https://issues.redhat.com/browse/OCPBUGS-30041
> 
>     * new machine/node labels and taint: the taint applied to the node is preventing spreading regular cluster workloads to the Outpost nodes, saving extra configuration like changing the default CSI, and also dedicating the AWS Oupost infrastructure to the user-workloads

- added these as additional reqs/limits
- reordered reqs/limits to try to better group items by topic and arrange in the order the reader encounters them in the setup process
- reworded callout 8 in the machine set YAML (I think it was sort of backwards; fixed that and explained relationship to workload toleration)

lmk what you think :)

mtulio reviewed (commented):
Thanks Jeana! Some NITs. Let me knwo what do you think.

jeana-redhat commented on modules/aws-outposts-requirements-limitations.adoc:
```diff
@@ -10,16 +10,21 @@ You can manage the resources on your Outpost similarly to those on a cloud-based
+
+* Due to the behavior of the upstream AWS cloud controller manager, you must include the `kubernetes.io/cluster/unmanaged` tag in the Outpost subnet configuration.
+Specifying this tag prevents the cloud controller manager from creating unsupported services in the Outpost subnet.
+This requirement is a workaround in {product-title} version {product-version}.
+For more information, see link:https://issues.redhat.com/browse/OCPBUGS-30041[OCPBUGS-30041].
```
I also wonder :sweat_smile: We use it in bug text for known issues but I will see if there is non-rel notes precedent

jeana-redhat replied:
seems to have precedent :)
![image](https://github.com/openshift/openshift-docs/assets/61474374/88b604cc-7343-4413-88b8-68c5d119c3fa)

jeana-redhat commented:
> Thanks Jeana! Some NITs. Let me knwo what do you think.

Good, thanks! Didn't take them all verbatim, so lmk if I have the meaning correct where I diverged.

mtulio commented:
/lgtm

jeana-redhat commented:
@alebedev87 PTAL when you have time

@yunjiang29 please check for QE :bow:

alebedev87 commented on modules/aws-outposts-requirements-limitations.adoc:
```diff
@@ -30,4 +36,8 @@ The Outpost has edge compute nodes, while the VPC has cloud-based compute nodes.
\ No newline at end of file
+To migrate persistent storage data for applications, you must perform a manual backup and restore operation.
+
+* AWS Outposts does not support AWS Network Load Balancers or AWS Classic Load Balancers.
+To manages Ingress objects for your edge compute resources, you must install the AWS Load Balancer Operator so that you can use AWS Application Load Balancers in the AWS Outposts environment.
```
It seems to me that there is no transition between "Outposts doesn't support NLB" and "Install ALBO to manage Ingress resources". May be something like - "Application Load Balancer is the only LB type supported on AWS Outposts. To be able to provision it you must 1) use Ingress resource and 2) install ALBO ".

alebedev87 commented on modules/aws-outposts-requirements-limitations.adoc:
```diff
@@ -30,4 +36,8 @@ The Outpost has edge compute nodes, while the VPC has cloud-based compute nodes.
+
+* AWS Outposts does not support AWS Network Load Balancers or AWS Classic Load Balancers.
+To manages Ingress objects for your edge compute resources, you must install the AWS Load Balancer Operator so that you can use AWS Application Load Balancers in the AWS Outposts environment.
+If your cluster contains both edge and cloud-based compute instances that share workloads, additional configuration is required.
```
No link to `Using the AWS Load Balancer Operator in an AWS VPC cluster extended into an Outpost` chapter?

jeana-redhat replied:
Unfortunately, the RH customer portal does not support links from anything in the `modules` directory :( Our usual workaround is to include an "Additional resources" list immediately after the module. I can add one with a link to the relevant content.

jeana-redhat commented:
Thanks @alebedev87! See what you think of that bullet now :)

alebedev87 commented:
/lgtm

jeana-redhat commented:
Thanks, Andrey! @yunjiang29 lmk if this all checks out from a QA perspective :slightly_smiling_face:

yunjiang29 commented:
/lgtm

dfitzmau commented:
/remove-label peer-review-needed

/label peer-review-in-progress

dfitzmau commented on modules/aws-outposts-machine-set.adoc:
```diff
@@ -165,7 +165,7 @@ spec:
 <6> Specifies the AWS region in which the Outpost availability zone exists.
 <7> Specifies the dedicated subnet for your Outpost.
-<8> Specifies a taint to prevent user workloads from being scheduled on nodes that have the `node-role.kubernetes.io/outposts` label.
+<8> Specifies a taint to prevent workloads from being scheduled on nodes that have the `node-role.kubernetes.io/outposts` label. To schedule user workloads in the Outpost, you must specify a corresponding toleration in the `Deployment` resource for your application.
```
Outpost node?

dfitzmau commented on modules/aws-outposts-requirements-limitations.adoc:
```diff
@@ -8,18 +8,24 @@
 You can manage the resources on your Outpost similarly to those on a cloud-based AWS cluster if you configure your {product-title} cluster to accommodate the following requirements and limitations:
 
-* To extend an {product-title} cluster on AWS into an Outpost, you must have installed the cluster into an existing VPC.
+* To extend an {product-title} cluster on AWS into an Outpost, you must have installed the cluster into an existing Amazon Virtual Private Cloud (VPC).
```
```suggestion
* To extend an {product-title} cluster on AWS into an AWS Outpost, you must have installed the cluster into an existing Amazon Virtual Private Cloud (VPC).
```

Add a link to dd.resoruces section that points to https://72291--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-vpc ?

dfitzmau replied:
Best to state AWS Outpost on first occurence in a module to show owndership to AWS and thereafter just use "Outpost"?

jeana-redhat replied:
The VPC install is already linked above (not in the changed module). Will add "AWS" to the first instance :)

dfitzmau commented on modules/aws-outposts-requirements-limitations.adoc:
```diff
@@ -8,18 +8,24 @@
+AWS Outposts do not support running service load balancers.
+To prevent the cloud controller manager from creating unsupported services in the Outpost subnet, you must include the `kubernetes.io/cluster/unmanaged` tag in the Outpost subnet configuration.
+This requirement is a workaround in {product-title} version {product-version}.
+For more information, see link:https://issues.redhat.com/browse/OCPBUGS-30041[OCPBUGS-30041].
```
I've not seen feature docs point to a bug. Can we document the steps in the doc instead of stating the bug? I see this in the event when the bug is resolved and who takes responsibility to check its status to prompt its removal from the doc.

jeana-redhat replied:
We do document the steps in the process, the bug link is just for additional reference

dfitzmau reviewed (commented):
Hi @jeana-redhat . Gerat work in getting the AWS Outposts docs to where it;s at. I added minor comments inline.

dfitzmau commented:
/remove-label peer-review-in-progress

/label peer-review-done

jeana-redhat commented:
/cherrypick enterprise-4.16

jeana-redhat commented:
/cherrypick enterprise-4.15
